ledger/src/blockstore_db.rs

```diff
@@ -952,10 +996,40 @@ pub struct WriteBatch<'a> {
     map: HashMap<&'static str, &'a ColumnFamily>,
 }
 
+pub enum ShredStorageType {
+    // Stores shreds under RocksDB's default compaction (level).
+    RocksLevel,
+    // (Experimental) Stores shreds under RocksDB's FIFO compaction which
+    // allows ledger store to reclaim storage more efficiently with
+    // lower I/O overhead.
+    RocksFifo,
+}
+
 pub struct BlockstoreOptions {
+    // The access type of blockstore. Default: PrimaryOnly
     pub access_type: AccessType,
+    // Whether to open a blockstore under a recovery mode. Default: None.
     pub recovery_mode: Option<BlockstoreRecoveryMode>,
+    // Whether to allow unlimited number of open files. Default: true.
     pub enforce_ulimit_nofile: bool,
+    // Determine how to store both data and coding shreds. Default: RocksLevel.
+    pub shred_storage_type: ShredStorageType,
+    // The maximum storage size for storing data shreds in column family
+    // [`cf::DataShred`].  Typically, data shreds contribute around 25% of the
+    // ledger store storage size if the RPC service is enabled, or 50% if RPC
+    // service is not enabled.
+    //
+    // Currently, this setting is only used when shred_storage_type is set to
+    // [`ShredStorageType::RocksFifo`].
+    pub shred_data_cf_size: u64,
+    // The maximum storage size for storing coding shreds in column family
+    // [`cf::CodeShred`].  Typically, coding shreds contribute around 20% of the
+    // ledger store storage size if the RPC service is enabled, or 40% if RPC
+    // service is not enabled.
+    //
+    // Currently, this setting is only used when shred_storage_type is set to
+    // [`ShredStorageType::RocksFifo`].
+    pub shred_code_cf_size: u64,
```

Comment:
Since these two fields are only necessary when FIFO compaction is enabled, how about a separate struct `FifoCompactionOptions`, and we have an optional field here: `fifo_compaction_options: Option<FifoCompactionOptions>`

Reply:
These two are from its parent PR (which is already merged.). I will incorporate the comments in a separate PR.